utils/cache.js

```diff
@@ -59,25 +69,87 @@ const cachePool = (opt = { maximumSize: CACHE_SIZE_MB }) => {
     }
   };
 
-  return { get, set, hits, cacheStore };
+  return { get, set, evict, hits, cacheStore };
+};
+/**
+ * A MultiMap implementation where each key maps to set of unique values.
+ * It internally uses Map to store keys and values and to save multiple values it uses Set. Map<string,Set<string>>
```

Comment:
- why not just store it as an array of strings instead of set?

Reply:
Sets help improve the complexity of `addCachedKey` and `removeCachedKey` methods. By using array both would have O(n) complexities. 

Let me know if I got this wrong.

Reply:
- going through the code I was only able to see `.get`, `.delete` operations used for `keyStore` which is a map, but we are not using the set which is stored inside this map to push / pop any values to/from it.

```js
const key = keyStore.get(key)
key.add(value)    // Not doing this anywhere
key.delete(value)    // Not doing this anywhere
```

- If we're not using the set stored inside the `keyStore` map what benefit does it provide storing it as a set and not an array?

Can you please point to the line of code where we're using this set?

Reply:
I want the cached keys to be unique, so if we were using an array we would have to iterate through the array to check if an element is already present. Sets can do the same with better complexity right?

```javascript

keyStore.get(modelKey).add(cachedKey);

```
https://github.com/Real-Dev-Squad/website-backend/pull/1283/files#diff-5ceb95acc7e3f0b7328eba1584f1eb987b65f01cb9c396d74a733d3a6b0565b9R101


Also I plan to evict only a cached key in a future PR 
 so will be using 'removeCachedKey' there which does `key.delete(value)`

Reply:
Thanks for the info closing this.